Add functions section to docs (#5271)
knative/docs: docs/snippets/proc-building-function.md

<!-- Snippet used in the following topics:
- /docs/getting-started/build-run-deploy-func.md
- /docs/functions/build-run-deploy-func.md
-->

> The `build` command uses the image name and the image registry to build a container image for your function.

Reviewer: ```suggestion
The `build` command uses the project name and the image registry name to construct a fully qualified container image name for your function.
```

> The `func.yaml` file is read to determine the image name and registry. If the function project has not previously been built, either the registry or the image name must be provided, and the image name is stored in the configuration `func.yaml` file.

Reviewer: ```suggestion
If the function project has not previously been built, you will be prompted to provide an image registry.
```
The user __may__ specify the registry (or image) on the command line. However, we have improved the UX so that if it is not known (either because it was not specified via `--registry` or the `FUNC_REGISTRY` environment variable is unset), the user will be prompted. E.g.

```
❯ ./func build -p foobar
A registry for function images is required. For example, 'docker.io/tigerteam'.
? Registry for function images:
```

Also, I'm not sure we really need to surface `func.yaml` yet. 🤷

=== "func"

>     If you are rebuilding a function by using a previously supplied image from a local `func.yaml` file, run the following command:

Reviewer: ```suggestion
    To build the function, run the following command:
```

    ```bash
    func build
    ```

>     If you ned to specify the registry or image for a new function build, run one of the following commands inside the project directory:
> 
>     ```bash
>     func build --registry <registry>
>     ```
> 
>     ```bash
>     func build --image <image-name>
>     ```
> 

Reviewer: ```suggestion
```
The flags are no longer required. Rather, the user will be prompted.

=== "kn func"

>     If you are rebuilding a function by using a previously supplied image from a local `func.yaml` file, run the following command:

Reviewer: ```suggestion
    To build the function, run the following command:
```

    ```bash
    kn func build
    ```

>     If you ned to specify the registry or image for a new function build, run one of the following commands inside the project directory:
> 
>     ```bash
>     kn func build --registry <registry>
>     ```
> 
>     ```bash
>     kn func build --image <image-name>
>     ```

Reviewer: ```suggestion
```
The flags are no longer required. Rather, the user will be prompted.
